codec/src/main/java/io/netty/handler/codec/compression/CompressionEncoder.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.ChannelPromiseNotifier;
+import io.netty.handler.codec.MessageToByteEncoder;
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.OneTimeTask;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Compresses a {@link ByteBuf} using one of compression algorithms.
+ */
+public abstract class CompressionEncoder extends MessageToByteEncoder<ByteBuf> {
+    /**
+     * Default delay time for close. See method {@link #closeDelay(int)}.
+     */
+    public static final int DEFAULT_CLOSE_DELAY = 10;
```

Comment:
Let's make this private just in case we use an interesting strategy that does not require this property at all.